pylib/tools/genhooks.py

```diff
@@ -67,6 +67,16 @@
         Your add-on can check filter_name to decide whether it should modify
         field_text or not before returning it.""",
     ),
+    Hook(
+        name="note_is_being_flushed",
+        args=["note: Note"],
+        doc="Allow to change a note before adding it to the database.",
+    ),
+    Hook(
+        name="card_is_being_flushed",
+        args=["card: Card"],
+        doc="Allow to change a card before adding it to the database.",
+    ),
```

Comment:
please use 'card_will_flush' and 'note_will_flush' for consistency with the other hook names

Reply:
I guess I don't really know what "flush" mean. I thought the process already started when the "flush" is called, and ends when the method "flush" returns.
But to be honest, I have no strong opinion on the hooks name. I just hope that I'll eventually be able to create correct hooks name without needing to change them after my PR

Reply:
The rule is pretty simple - if a hook is before some action, use 'will'. If the hook is after some action, use 'did'. :-)

'flush' comes from 'flush out', as in send/push out any pending changes that have not yet been saved to disk

It's probably not worth changing here (and I frequently make the same mistake myself), but in new methods, it'd be good if we could snake_case instead of camelCaps.

Could I trouble you to resolve the conflict that merging your earlier PR has caused? Thanks for that by the way :-)

Reply:
You mean, for _preFlush ? 
I wanted cards and notes to be consistent in method name. 
Otherwise I used snake. Its make develop wich transform it into camlCase

Reply:
Yes, I meant _preFlush. Again, not a big deal, but better to use snake_case in future methods you add. Thanks for the patch!